### PR TITLE
[website] Update cli-tools document

### PR DESCRIPTION
**Pull request by yuruguo**
### Motivation
The related options of cli tools are not updated and synchronized in time，which causes some important options to be invisible on the document / website.
[reference-cli-tools](https://pulsar.apache.org/docs/en/reference-cli-tools/)
[performance-pulsar-perf](https://pulsar.apache.org/docs/en/performance-pulsar-perf/)

### Modifications
- update `reference-cli-tools.md`(include `master`, `2.8.x`, `2.7.x`, `2.6.x`, `2.5.x`, `2.4.x`, `2.3.x`)
- update `performance-pulsar-perf.md`(include `master`, `2.8.x`, `2.7.x`, `2.6.x`)
- `reference-pulsar-admin.md` is deprecated and not updated anymore so do nothing

### CODE REVIEW

**Comment by yuruguo** [comment]
@tuteng @Anonymitaet PTAL, thx!

File: site2/docs/performance-pulsar-perf.md [hunk 1]
```diff
@@ -42,34 +42,48 @@ The following table lists configuration options available for the `pulsar-perf p
 
 | Option | Description | Default value|
 |----|----|----|
+| access-mode | Set the producer access mode, valid values: [Shared, Exclusive, WaitForExclusive]. | Shared |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:45]:
```suggestion
| access-mode | Set the producer access mode. Valid values are `Shared`, `Exclusive`, and `WaitForExclusive`]. | Shared |
```

File: site2/docs/performance-pulsar-perf.md [hunk 2]
```diff
@@ -42,34 +42,48 @@ The following table lists configuration options available for the `pulsar-perf p
...
+| admin-url | Set the pulsar admin url. | N/A |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:46]:
```suggestion
| admin-url | Set the Pulsar admin URL. | N/A |
```

File: site2/docs/performance-pulsar-perf.md [hunk 3]
```diff
@@ -42,34 +42,48 @@ The following table lists configuration options available for the `pulsar-perf p
...
 | encryption-key-name | Set the name of the public key used to encrypt the payload. | N/A |
 | encryption-key-value-file | Set the file which contains the public key used to encrypt the payload. | N/A |
 | exit-on-failure | Configure whether to exit from the process on publish failure. | false |
+| format-class | Set the custom formatter class name. | org.apache.pulsar.testclient.DefaultMessageFormatter |
+| format-payload | Configure whether to format %i as a message index in the stream from producer and/or %t as the timestamp nanoseconds. | false |
 | help | Configure the help message. | false |
 | max-connections | Set the maximum number of TCP connections to a single broker. | 100 |
 | max-outstanding | Set the maximum number of outstanding messages. | 1000 |
 | max-outstanding-across-partitions | Set the maximum number of outstanding messages across partitions. | 50000 |
+| message-key-generation-mode | Set the generation mode of message key, valid options: [autoIncrement, random]. | N/A |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:67]: same comment

File: site2/docs/performance-pulsar-perf.md [hunk 4]
```diff
@@ -42,34 +42,48 @@ The following table lists configuration options available for the `pulsar-perf p
...
 | rate | Set the publish rate of messages across topics. | 100 |
+| send-timeout | Set the sendTimeout. | 0 |
+| separator | Set the separator between the topic and topic number. | - |
 | service-url | Set the Pulsar service URL. | |
 | size | Set the message size. | 1024 bytes |
 | stats-interval-seconds | Set the statistics interval. If it is set to 0, statistics is disabled. | 0 |
 | test-duration | Set the test duration. If it is set to 0, it keeps publishing tests. | 0s |
 | trust-cert-file | Set the path for the trusted TLS certificate file. | <empty string> |
 | warmup-time | Set the warm-up time. | 1s |
+| tls-allow-insecure | Set the allow insecure TLS connection. | N/A |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:86]:
```suggestion
| tls-allow-insecure | Set the allowed insecure TLS connection. | N/A |
```

File: site2/docs/performance-pulsar-perf.md [hunk 5]
```diff
@@ -109,21 +123,33 @@ The following table lists configuration options available for the `pulsar-perf c
 | acks-delay-millis | Set the acknowledgment grouping delay in milliseconds. | 100 ms |
 | auth-params | Set the authentication parameters, whose format is determined by the implementation of the `configure` method in the authentication plugin class, such as "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}". | N/A |
 | auth_plugin | Set the authentication plugin class name. | N/A |
+| auto_ack_chunk_q_full | Configure whether to auto ack for oldest message on queue is full. | false |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:126]: Sorry, this is a grammar mistake in this sentence, can you rewrite it?

File: site2/docs/performance-pulsar-perf.md [hunk 6]
```diff
@@ -109,21 +123,33 @@ The following table lists configuration options available for the `pulsar-perf c
...
+| listener-name | Set the listener name for the broker. | N/A |
+| acks-delay-millis | Set the acknowledgements grouping delay in millis. | 100 |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:128]:
```suggestion
| acks-delay-millis | Set the acknowledgments grouping delay (in milliseconds). | 100 |
```
Do you mean this

File: site2/docs/performance-pulsar-perf.md [hunk 7]
```diff
@@ -109,21 +123,33 @@ The following table lists configuration options available for the `pulsar-perf c
...
+| batch-index-ack | Enable or disable the batch index acknowledgment. | false |
+| busy-wait | Enable or disable Busy-Wait on the Pulsar client. | false |
 | conf-file | Set the configuration file. | N/A |
 | encryption-key-name | Set the name of the public key used to encrypt the payload. | N/A |
 | encryption-key-value-file | Set the file which contains the public key used to encrypt the payload. | N/A |
 | help | Configure the help message. | false |
+| expire_time_incomplete_chunked_messages | Set the expire time in ms for incomplete chunk messages. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:135]:
```suggestion
| expire_time_incomplete_chunked_messages | Set the expiration time for incomplete chunk messages (in milliseconds). | 0 |
```
Do you mean this?

File: site2/docs/performance-pulsar-perf.md [hunk 8]
```diff
@@ -109,21 +123,33 @@ The following table lists configuration options available for the `pulsar-perf c
...
 | rate | Simulate a slow message consumer (rate in msg/s). | 0.0 |
 | receiver-queue-size | Set the size of the receiver queue. | 1000 |
+| receiver-queue-size-across-partitions | Set the max total size of the receiver queue across partitions. | 50000 |
 | replicated | Configure whether the subscription status should be replicated. | false |
 | service-url | Set the Pulsar service URL. | |
 | stats-interval-seconds | Set the statistics interval. If it is set to 0, statistics is disabled. | 0 |
 | subscriber-name | Set the subscriber name prefix. | sub |
 | subscription-type | Set the subscription type. <li> Exclusive <li> Shared <li> Failover <li> Key_Shared | Exclusive |
 | trust-cert-file | Set the path for the trusted TLS certificate file. | <empty string> |
+| test-duration | Set the test duration in secs. If <= 0 it will keep consuming. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:152]:
```suggestion
| test-duration | Set the test duration (in seconds). If the value is 0 or smaller than 0, it keeps consuming messages. | 0 |
```
Do you mean this?

File: site2/docs/reference-cli-tools.md [hunk 9]
```diff
@@ -321,6 +332,8 @@ Options
 |`-s`, `--separator`|Character to split messages string with.|","|
 |`-k`, `--key`|Message key to add|key=value string, like k1=v1,k2=v2.|
 |`-p`, `--properties`|Properties to add. If you want to add multiple properties, use the comma as the separator, e.g. `k1=v1,k2=v2`.| |
+|`-ekn`, `--encryption-key-name`|The public key name to encrypt payload.| |
+|`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload, for example file:///path/to/public.key or data:application/x-pem-file;base64,*****.| |
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:336]:
```suggestion
|`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload. For example, `file:///path/to/public.key or data:application/x-pem-file;base64,*****`.| |
```

File: site2/docs/reference-cli-tools.md [hunk 10]
```diff
@@ -347,6 +361,8 @@ Options
 |`-ac`, `--auto_ack_chunk_q_full`|Auto ack for the oldest message in consumer's receiver queue if the queue full.|false|
 |`--hide-content`|Do not print the message to the console.|false|
 |`-st`, `--schema-type`|Set the schema type. Use `auto_consume` to dump AVRO and other structured data types. Possible values: bytes, auto_consume.|bytes|
+|`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload, for example file:///path/to/public.key or data:application/x-pem-file;base64,*****.| |
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:364]: same

File: site2/docs/reference-cli-tools.md [hunk 11]
```diff
@@ -429,25 +446,35 @@ Options
 
 |Flag|Description|Default|
 |---|---|---|
-|`--auth_params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
+|`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:449]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 12]
```diff
@@ -429,25 +446,35 @@ Options
...
+|`-ac`, `--auto_ack_chunk_q_full`|Auto ack for oldest message on queue is full|false|
 |`--listener-name`|Listener name for the broker||
 |`--acks-delay-millis`|Acknowledgements grouping delay in millis|100|
-|`-k`, `--encryption-key-name`|The private key name to decrypt payload||
+|`--batch-index-ack`|Enable or disable the batch index acknowledgment|false|
+|`-bw`, `--busy-wait`|Enable Busy-Wait on the Pulsar client|false|
 |`-v`, `--encryption-key-value-file`|The file which contains the private key to decrypt payload||
 |`-h`, `--help`|Help message|false|
 |`--conf-file`|Configuration file||
+|`-e`, `--expire_time_incomplete_chunked_messages`|Expire time in ms for incomplete chunk messages|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:459]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 13]
```diff
@@ -429,25 +446,35 @@ Options
...
 |`-q`, `--receiver-queue-size`|Size of the receiver queue|1000|
+|`-p`, `--receiver-queue-size-across-partitions`|Max total size of the receiver queue across partitions|50000|
+|`--replicated`|Whether the subscription status should be replicated|false|
 |`-u`, `--service-url`|Pulsar service URL||
 |`-i`, `--stats-interval-seconds`|Statistics interval seconds. If 0, statistics will be disabled|0|
 |`-s`, `--subscriber-name`|Subscriber name prefix||
 |`-ss`, `--subscriptions`|A list of subscriptions to consume on (e.g. sub1,sub2)|sub|
 |`-st`, `--subscription-type`|Subscriber type. Possible values are Exclusive, Shared, Failover, Key_Shared.|Exclusive|
 |`-sp`, `--subscription-position`|Subscriber position. Possible values are Latest, Earliest.|Latest|
+|`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep consuming|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:477]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 14]
```diff
@@ -464,29 +491,45 @@ Options
 
 |Flag|Description|Default|
 |---|---|---|
-|`--auth_params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
+|`-am`, `--access-mode`|Producer access mode, valid values: [Shared, Exclusive, WaitForExclusive]|Shared|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:494]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 15]
```diff
@@ -464,29 +491,45 @@ Options
...
+|`-au`, `--admin-url`|Pulsar Admin URL||
+|`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:496]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 16]
```diff
@@ -464,29 +491,45 @@ Options
...
 |`--auth_plugin`|Authentication plugin class name||
 |`--listener-name`|Listener name for the broker||
 |`-b`, `--batch-time-window`|Batch messages in a window of the specified number of milliseconds|1|
+|`-bb`, `--batch-max-bytes`|Maximum number of bytes per batch|4194304|
+|`-bm`, `--batch-max-messages`|Maximum number of messages per batch|1000|
+|`-bw`, `--busy-wait`|Enable Busy-Wait on the Pulsar client|false|
+|`-ch`, `--chunking`|Should split the message and publish in chunks if message size is larger than allowed max size|false|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:503]:
```suggestion
|`-ch`, `--chunking`|Split the message and publish in chunks if the message size is larger than allowed max size|false|
```

File: site2/docs/reference-cli-tools.md [hunk 17]
```diff
@@ -464,29 +491,45 @@ Options
...
 |`-v`, `--encryption-key-value-file`|The file which contains the public key to encrypt payload||
+|`-ef`, `--exit-on-failure`|Exit from the process on publish failure|false|
+|`-fc`, `--format-class`|Custom Formatter class name|org.apache.pulsar.testclient.DefaultMessageFormatter|
+|`-fp`, `--format-payload`|Format %i as a message index in the stream from producer and/or %t as the timestamp nanoseconds|false|
 |`-h`, `--help`|Help message|false|
 |`-c`, `--max-connections`|Max number of TCP connections to a single broker|100|
 |`-o`, `--max-outstanding`|Max number of outstanding messages|1000|
 |`-p`, `--max-outstanding-across-partitions`|Max number of outstanding messages across partitions|50000|
-|`-m`, `--num-messages`|Number of messages to publish in total. If set to 0, it will keep publishing.|0|
+|`-mk`, `--message-key-generation-mode`|The generation mode of message key, valid options: [autoIncrement, random]||
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:516]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 18]
```diff
@@ -464,29 +491,45 @@ Options
...
+|`-ioThreads`, `--num-io-threads`|Set the number of threads to be used for handling connections to brokers|1|
+|`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing.|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:518]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 19]
```diff
@@ -464,29 +491,45 @@ Options
...
 |`-n`, `--num-producers`|The number of producers (per topic)|1|
+|`-threads`, `--num-test-threads`|Number of test threads|1|
 |`-t`, `--num-topic`|The number of topics|1|
+|`-np`, `--partitions`|Create partitioned topics with the given number of partitions, set 0 to not try to create the topic||
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:522]:
```suggestion
|`-np`, `--partitions`|Create partitioned topics with the given number of partitions. Setting this value to 0 means not trying to create a topic||
```

File: site2/docs/reference-cli-tools.md [hunk 20]
```diff
@@ -464,29 +491,45 @@ Options
...
 |`-e`, `--payload-delimiter`|The delimiter used to split lines when using payload from a file|\n|
+|`-pn`, `--producer-name`|Producer Name||
 |`-r`, `--rate`|Publish rate msg/s across topics|100|
+|`--send-timeout`|Set the sendTimeout|0|
+|`--separator`|Separator between the topic and topic number|-|
 |`-u`, `--service-url`|Pulsar service URL||
 |`-s`, `--size`|Message size (in bytes)|1024|
 |`-i`, `--stats-interval-seconds`|Statistics interval seconds. If 0, statistics will be disabled.|0|
-|`-time`, `--test-duration`|Test duration in secs. If set to 0, it will keep publishing.|0|
+|`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep publishing.|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:532]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 21]
```diff
@@ -501,20 +544,23 @@ $ pulsar-perf read options
 ```
 
 Options
+
 |Flag|Description|Default|
 |---|---|---|
-|`--auth_params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
+|`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:550]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 22]
```diff
@@ -501,20 +544,23 @@ $ pulsar-perf read options
...
 |`-h`, `--help`|Help message|false|
 |`-c`, `--max-connections`|Max number of TCP connections to a single broker|100|
+|`-ioThreads`, `--num-io-threads`|Set the number of threads to be used for handling connections to brokers|1|
 |`-t`, `--num-topics`|The number of topics|1|
 |`-r`, `--rate`|Simulate a slow message reader (rate in msg/s)|0|
 |`-q`, `--receiver-queue-size`|Size of the receiver queue|1000|
 |`-u`, `--service-url`|Pulsar service URL||
 |`-m`, `--start-message-id`|Start message id. This can be either 'earliest', 'latest' or a specific message id by using 'lid:eid'|earliest|
 |`-i`, `--stats-interval-seconds`|Statistics interval seconds. If 0, statistics will be disabled.|0|
+|`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep consuming|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:563]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 23]
```diff
@@ -528,19 +574,20 @@ $ pulsar-perf websocket-producer options
 ```
 
 Options
+
 |Flag|Description|Default|
 |---|---|---|
-|`--auth_params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
+|`--auth-params`|Authentication parameters, whose format is determined by the implementation of method `configure` in authentication plugin class, for example "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}.||
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:580]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 24]
```diff
@@ -528,19 +574,20 @@ $ pulsar-perf websocket-producer options
...
 |`--auth_plugin`|Authentication plugin class name||
 |`--conf-file`|Configuration file||
 |`-h`, `--help`|Help message|false|
-|`-m`, `--num-messages`|Number of messages to publish in total. If 0, it will keep publishing|0|
+|`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:584]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 25]
```diff
@@ -528,19 +574,20 @@ $ pulsar-perf websocket-producer options
...
 |`-t`, `--num-topic`|The number of topics|1|
 |`-f`, `--payload-file`|Use payload from a file instead of empty buffer||
 |`-u`, `--proxy-url`|Pulsar Proxy URL, e.g., "ws://localhost:8080/"||
 |`-r`, `--rate`|Publish rate msg/s across topics|100|
 |`-s`, `--size`|Message size in byte|1024|
-|`-time`, `--test-duration`|Test duration in secs. If 0, it will keep publishing|0|
+|`-time`, `--test-duration`|Test duration in secs. If <=0, it will keep publishing|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:590]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 26]
```diff
@@ -560,11 +608,11 @@ Options
 |`-h`, `--help`|Help message|false|
 |`-c`, `--max-connections`|Max number of TCP connections to a single bookie|1|
 |`-o`, `--max-outstanding`|Max number of outstanding requests|1000|
-|`-m`, `--num-messages`|Number of messages to publish in total. If 0, it will keep publishing|0|
+|`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing|0|
 |`-t`, `--num-topic`|Number of managed ledgers|1|
 |`-r`, `--rate`|Write rate msg/s across managed ledgers|100|
 |`-s`, `--size`|Message size in byte|1024|
-|`-time`, `--test-duration`|Test duration in secs. If 0, it will keep publishing|0|
+|`-time`, `--test-duration`|Test duration in secs. If <= 0, it will keep publishing|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:615]: same comment as previously

File: site2/docs/reference-cli-tools.md [hunk 27]
```diff
@@ -560,11 +608,11 @@ Options
 |`-h`, `--help`|Help message|false|
 |`-c`, `--max-connections`|Max number of TCP connections to a single bookie|1|
 |`-o`, `--max-outstanding`|Max number of outstanding requests|1000|
-|`-m`, `--num-messages`|Number of messages to publish in total. If 0, it will keep publishing|0|
+|`-m`, `--num-messages`|Number of messages to publish in total. If <= 0, it will keep publishing|0|
```
Comment (Anonymitaet) [inline site2/docs/reference-cli-tools.md:611]: same comment as previously

**Review by Anonymitaet — CHANGES_REQUESTED** [review]

**Comment by yuruguo** [comment]
Thanks for @Anonymitaet guidance, I have modified it according to your suggestion, PTAL again

File: site2/docs/performance-pulsar-perf.md (hunk 5 above)
Comment (yuruguo) [reply site2/docs/performance-pulsar-perf.md]: Okay

**Review by Anonymitaet — APPROVED** [review]

**Comment by yuruguo** [comment]
> Currently, after pulsar 2.8, most of the tools' documentation is automatically generated by the tool, and only a few of them are not yet covered by the automation tool, so my suggestion is to remove them from this documentation and add a link https://pulsar.apache.org/tools/pulsar-perf/2.9.0-SNAPSHOT/ to them for those already generated by the tool, such as pulsar-perf, etc. This will save us a lot of work, and the documentation will be correct will also be improved. @Anonymitaet @yuruguo

Thanks  for your reply @tuteng .
It is a very profitable thing to generate tools' documents through automated tools and provide special links. I found that `pulsar-admin` has been maintained in this way [since version `2.6`](https://pulsar.apache.org/tools/pulsar-admin/). But I have a few questions to confirm:
Do these cli-tools like `pulsar-client`, `pulsar-perf` and `pulsar` also use same method only from version `2.9`?
If yes, do we still need to manually update the previous version such as `2.8.x`, `2.7.x` or even earlier (ps: It‘s very important to maintain the consistency of the documentation, because from my experience in using pulsar, I started with the official documentation and core tools )? I am willing to continue to do this work and have manually updated part of the previous version(including `2.8.2`, `2.8.1`, `2.8.0`, `2.7.3`, `2.7.2`, `2.7.1` and `2.7.0`) although I haven't pushed it yet @tuteng @Anonymitaet .
If not, we may not need to manually update the relevant documents of the previous version, then this pr may be closed.

**Comment by yuruguo** [comment]
I have update docs about 2.7.x(`2.7.0`/`2.7.1`/`2.7.2`/`2.7.3`) and 2.8.x(`2.8.0`/`2.8.1`/`2.8.2`), PTAL @Anonymitaet @tuteng

File: site2/docs/performance-pulsar-perf.md [hunk 28]
```diff
@@ -109,22 +123,34 @@ The following table lists configuration options available for the `pulsar-perf c
...
 | auth_plugin | Set the authentication plugin class name. | N/A |
+| auto_ack_chunk_q_full | Configure whether to automatically ack for the oldest message in receiver queue if the queue is full. | false |
+| listener-name | Set the listener name for the broker. | N/A |
+| batch-index-ack | Enable or disable the batch index acknowledgment. | false |
+| busy-wait | Enable or disable Busy-Wait on the Pulsar client. | false |
 | conf-file | Set the configuration file. | N/A |
 | encryption-key-name | Set the name of the public key used to encrypt the payload. | N/A |
 | encryption-key-value-file | Set the file which contains the public key used to encrypt the payload. | N/A |
 | help | Configure the help message. | false |
+| expire_time_incomplete_chunked_messages | Set the expiration time for incomplete chunk messages (in milliseconds). | 0 |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:134]: underline (_) or hyphen (-)?

File: site2/docs/performance-pulsar-perf.md [hunk 29]
```diff
@@ -109,22 +123,34 @@ The following table lists configuration options available for the `pulsar-perf c
 | acks-delay-millis | Set the acknowledgment grouping delay in milliseconds. | 100 ms |
 | auth-params | Set the authentication parameters, whose format is determined by the implementation of the `configure` method in the authentication plugin class, such as "key1:val1,key2:val2" or "{"key1":"val1","key2":"val2"}". | N/A |
 | auth_plugin | Set the authentication plugin class name. | N/A |
+| auto_ack_chunk_q_full | Configure whether to automatically ack for the oldest message in receiver queue if the queue is full. | false |
```
Comment (Anonymitaet) [inline site2/docs/performance-pulsar-perf.md:126]: underline (_) or hyphen (-)?

Comment (Anonymitaet) [reply site2/docs/performance-pulsar-perf.md:126]: Please check all occurencies.

Comment (yuruguo) [reply site2/docs/performance-pulsar-perf.md:126]: @Anonymitaet Both `expire_time_incomplete_chunked_messages` and `auto_ack_chunk_q_full` use underline, they come from here
https://github.com/apache/pulsar/blob/f342c5d893527b5daec59cbc67abd3dbfdb02020/pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceConsumer.java#L145-L151

Comment (Anonymitaet) [reply site2/docs/performance-pulsar-perf.md:126]: OK

**Comment by yuruguo** [comment]
I have updated the remaining part of `cli-tools` document (from `2.3.x` to `2.6.x`), so far all versions of `cli-tools` document have been updated, although this is a relatively time-consuming and energy-consuming work, but a complete document can help users better use Pulsar. PTAL @Anonymitaet, thx!

https://github.com/apache/pulsar/pull/11980#issue-991942058

**Pull request merged**